homeassistant/components/somfy/climate.py

```diff
@@ -0,0 +1,214 @@
+"""Support for Somfy Thermostat."""
+
+import logging
+from typing import Any, Dict, List, Optional
+
+from pymfy.api.devices.category import Category
+from pymfy.api.devices.thermostat import (
+    DurationType,
+    HvacState,
+    RegulationState,
+    TargetMode,
+    Thermostat,
+)
+
+from homeassistant.components.climate import ClimateEntity
+from homeassistant.components.climate.const import (
+    CURRENT_HVAC_COOL,
+    CURRENT_HVAC_HEAT,
+    CURRENT_HVAC_IDLE,
+    HVAC_MODE_AUTO,
+    HVAC_MODE_COOL,
+    HVAC_MODE_HEAT,
+    PRESET_AWAY,
+    PRESET_HOME,
+    PRESET_SLEEP,
+    SUPPORT_PRESET_MODE,
+    SUPPORT_TARGET_TEMPERATURE,
+)
+from homeassistant.const import ATTR_BATTERY_LEVEL, ATTR_TEMPERATURE, TEMP_CELSIUS
+
+from . import SomfyEntity
+from .const import API, COORDINATOR, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+SUPPORTED_CATEGORIES = {Category.HVAC.value}
+
+PRESET_FROST_GUARD = "Frost Guard"
+PRESET_GEOFENCING = "Geofencing"
+PRESET_MANUAL = "Manual"
+
+PRESETS_MAPPING = {
+    TargetMode.AT_HOME: PRESET_HOME,
+    TargetMode.AWAY: PRESET_AWAY,
+    TargetMode.SLEEP: PRESET_SLEEP,
+    TargetMode.MANUAL: PRESET_MANUAL,
+    TargetMode.GEOFENCING: PRESET_GEOFENCING,
+    TargetMode.FROST_PROTECTION: PRESET_FROST_GUARD,
+}
+REVERSE_PRESET_MAPPING = {v: k for k, v in PRESETS_MAPPING.items()}
+
+HVAC_MODES_MAPPING = {HvacState.COOL: HVAC_MODE_COOL, HvacState.HEAT: HVAC_MODE_HEAT}
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Somfy climate platform."""
+
+    def get_thermostats():
+        """Retrieve thermostats."""
+        domain_data = hass.data[DOMAIN]
+        coordinator = domain_data[COORDINATOR]
+        api = domain_data[API]
+
+        return [
+            SomfyClimate(coordinator, device_id, api)
+            for device_id, device in coordinator.data.items()
+            if SUPPORTED_CATEGORIES & set(device.categories)
+        ]
+
+    async_add_entities(await hass.async_add_executor_job(get_thermostats))
+
+
+class SomfyClimate(SomfyEntity, ClimateEntity):
+    """Representation of a Somfy thermostat device."""
+
+    def __init__(self, coordinator, device_id, api):
+        """Initialize the Somfy device."""
+        super().__init__(coordinator, device_id, api)
+        self._climate = None
+        self._create_device()
+
+    def _create_device(self):
+        """Update the device with the latest data."""
+        self._climate = Thermostat(self.device, self.api)
+
+    @property
+    def supported_features(self) -> int:
+        """Return the list of supported features."""
+        return SUPPORT_TARGET_TEMPERATURE | SUPPORT_PRESET_MODE
+
+    @property
+    def device_state_attributes(self) -> Dict[str, Any]:
+        """Return the state attributes of the device."""
+        return {ATTR_BATTERY_LEVEL: self._climate.get_battery()}
```

Comment:
Battery level should be a separate sensor entity. Please remove this state attribute.

Reply:
So should we update the [documentation](https://developers.home-assistant.io/docs/core/entity#standard-attributes)?

Reply:
Yes, please.

Reply:
One more question sorry.

Does it means the end user will not see anymore the battery icon beside the entities?

Do we plan to add a similar feature for devices?

Reply:
The battery report in the device list is taken from a sensor entity with device class `DEVICE_CLASS_BATTERY`.

Reply:
Good to know. Thx.